### PR TITLE
Ensure proper behavior when slist expires

### PR DESCRIPTION
**Pull request by jerryblakley**
connects to pantheon-systems/platform#4

1. On GETADDRINFO_FAILURE we weren't capturing the NULL from create_new_addr_table, and this ultimately led to a NULL-pointer exception.
2. We weren't actually creating an updated resolve_slist on the 2-minute timeout. (It is still not clear that the new list, not 'forced', actually ever takes effect; still need to delve into that.)

### CODE REVIEW

File: src/session.c [hunk 1]
```diff
@@ -1243,8 +1243,11 @@ static CURL *get_session(bool tmp_session) {
 
     GHashTable *addr_table;
     addr_table = create_new_addr_table();
+    // On getaddrinfo failure, NULL gets returned; pass it through
+    if (addr_table == NULL) return NULL;
+
     status = needs_new_session(addr_table);
-    if (status == BOTH || status == SESSION) {
+    if (status != NEITHER) {
```
Comment (kibra) [inline src/session.c]: What is `NEITHER` ?

Comment (jerryblakley) [inline src/session.c]: I'm going to simplify the logic. I need to distinguish between 'SESSION' and 'SLIST' to decide whether or not to cleanup the old session, but that distinction does not need to be made elsewhere. Update in a moment

File: src/session.c [hunk 2]
```diff
@@ -1212,46 +1201,56 @@ static CURL *update_session(GHashTable *addr_table, bool tmp_session, enum slist
...
-    enum new_session_new_slist status;
     static const char * funcname = "get_session";
 
     GHashTable *addr_table;
     addr_table = create_new_addr_table();
-    status = needs_new_session(addr_table);
-    if (status == BOTH || status == SESSION) {
-        session = update_session(addr_table, tmp_session, status);
+    // On getaddrinfo failure, NULL gets returned; pass it through
+    if (addr_table == NULL) return NULL;
```
Comment (kibra) [inline src/session.c]: Right, I see.

**Comment by kibra** [comment]
+1

:fireworks:

**Pull request merged**